lib/recurly/client.php

```diff
@@ -141,6 +149,14 @@ private function _sendRequest($method, $uri, $data = '')
       curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $method);
     }
 
+    if(Recurly_Client::$logger){
+      Recurly_Client::$logger->info('Send request to Recurly', array('method' => $method, 'uri' => $uri));
+      Recurly_Client::$logger->debug('Send request with headers', array('headers' => $headers));
+      if(!empty($data)){
+        Recurly_Client::$logger->debug('Recurly request body', array('body' => $data));
+      }
+    }
```

Comment:
It makes more sense to me to have only one debug call with the body and headers, but I had other motives.

For our project, I wanted PSR-3 logging to track our Recurly transactions.  I wanted this patch to get into the PHP client, and I thought, "Well, [here's how Recurly did it in Ruby](https://github.com/recurly/recurly-client-ruby/blob/master/lib/recurly/api/net_http_adapter.rb#L65). I'll mimic that."  It seemed like the best way to get the feature in and start the conversation.

Regarding `null`, I also think it makes sense that the log shows `null` when the body is `null`. As a side note, in our production code, I do the *awful* thing of using the STFU operator `@` with  some nice benefits.  If a property of an object does not exist, no additional error is thrown and the logs show `null` for the output of that property value.

```
$logger->info('Recording the value of adjustment id', array('adjustment_id' => @$adjustment_id));
```

The missing $adjustment_id variable will be cast as `null` and error for using an undefined variable will be suppressed.